addons/website/static/src/components/configurator/configurator.js

```diff
@@ -617,6 +618,30 @@ async function applyConfigurator(self, themeName) {
         self.env.router.navigate({to: 'CONFIGURATOR_PALETTE_SELECTION_SCREEN'});
         return;
     }
+
+    async function attemptConfiguratorApply(data, retryCount = 0) {
+        try {
+            const resp = await self.rpc({
```

Comment:
```suggestion
            return await self.rpc({
```

Reply:
`return await` is probably always wrong. When you are in an `async` function (which is the case here since there is an `await`), the return will always automatically be a `Promise` instance. If you write `return 3`, the return won't be 3 but a `Promise` instance which resolves to `3`. So if you write `return await x`, it means that won't the value resolved from x... but you return it as a `Promise` anyway, so it's the same as `return x`.

Reply:
In this case, the `await` is needed - so that the error gets raised, and therefore chaught by the `catch` block.
I tried rewriting it as a `return self.rpc.(...).guardedCatch(...)` but each error reaching the guarded catch then gets displayed instead of only the last one - and since the purpose of this task is to make it look as if it was flawless, I kept the `try { return await } catch { ... }` structure.

Reply:
Oh ok, good point, I guess there is an usecase for `return await` after all, my bad :+1: (don't know why the `guardedCatch` implementation would display the error each loop step though but not worth losing time over this :+1:)

Reply:
It is actually because `guardedCarch` always `return Promise.reject(reason);`, it lacks a way for the handler function to indicate that it should not do that.

Reply:
Ok, I am not a fan of `guardedCatch` anyway, it has many flaws :wink: